wdl2cwl/main.py

```diff
@@ -277,99 +282,118 @@ def load_wdl_workflow(self, obj: WDL.Tree.Workflow) -> cwl.Workflow:
-                                inputs_from_call[key] = (
-                                    input_expr.replace(".", "/"),
-                                    {},
-                                )
-                wf_step_inputs: List[cwl.WorkflowStepInput] = []
-                for inp in cwl_callee_inputs:
-                    call_inp_id = f"{call.name}.{inp.id}"
-                    source_str, extras = inputs_from_call.get(
-                        cast(str, inp.id), (call_inp_id, {})
-                    )
-
-                    if inp.id not in input_defaults:
-                        wf_step_inputs.append(
-                            cwl.WorkflowStepInput(
-                                id=inp.id, source=source_str, **extras
-                            )
-                        )
-                    else:
-                        wf_step_inputs.append(
-                            cwl.WorkflowStepInput(
-                                id=inp.id, default=source_str, **extras
-                            )
-                        )
-                wf_step_outputs = (
-                    [
-                        cwl.WorkflowStepOutput(id=output.name)
-                        for output in callee.outputs
-                    ]
-                    if callee.outputs
-                    else []
-                )
-                wf_step_run = self.load_wdl_objects(callee)
-                wf_step = cwl.WorkflowStep(
-                    wf_step_inputs,
-                    id=call.name,
-                    run=wf_step_run,
-                    out=wf_step_outputs,
-                )
-                wf_steps.append(wf_step)
+            with WDLSourceLine(body_part, ConversionException):
+                if isinstance(body_part, WDL.Tree.Call):
+                    wf_steps.append(self.get_workflow_call(body_part))
+                elif isinstance(body_part, WDL.Tree.Scatter):
+                    is_scatter_present = True
+                    wf_steps.extend(self.get_workflow_scatter(body_part))
+        requirements = [cwl.ScatterFeatureRequirement()] if is_scatter_present else None
 
         return cwl.Workflow(
             id=wf_name,
             cwlVersion="v1.2",
             doc=wf_description,
+            requirements=requirements,
             inputs=inputs,
             steps=wf_steps,
             outputs=outputs,
         )
 
+    def get_workflow_scatter(self, scatter: WDL.Tree.Scatter) -> List[cwl.WorkflowStep]:
+        """Get the CWL Workflow Step equivalent of a list of WDL Scatter Object."""
+        scatter_steps: List[cwl.WorkflowStep] = []
+        for body_obj in scatter.body:
+            with WDLSourceLine(body_obj, ConversionException):
+                if isinstance(body_obj, WDL.Tree.Call):
+                    wf_step = self.get_workflow_call(body_obj)
+                wf_step.scatter = self.scatter_names.pop()
```

Comment:
I think if the `isinstance` above fails, this would result in a... `NameNotFound` error? Can't recall the exception for variables not yet declared. But if the `if` above can be `False`, we have to be more defensive here IMO.

Reply:
I am yet to encounter anything other than a WDL.Tree.Call inside a scatter object but I believe it's possible. The `with WDLSourceLine` handles the case when it is not.